assert/assertions_test.go

```diff
@@ -2970,6 +2970,49 @@ func TestEventuallyWithT_ReturnsTheLatestFinishedConditionErrors(t *testing.T) {
 	Len(t, mockT.errors, 2)
 }
 
+// Check that a long running condition doesn't block Eventually.
+// See issue 805 (and its long tail of following issues)
+func TestEventuallyTimeout(t *testing.T) {
```

Comment:
This test isn't new; I just moved it so that all of the `Eventually*` tests are grouped together rather than having the `Never*` tests in between them